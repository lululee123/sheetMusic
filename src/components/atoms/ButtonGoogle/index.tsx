import React from 'react';
import classnames from 'classnames';

import GoogleIcon from 'images/icon/google.inline.svg';

import ButtonWithIcon from 'components/atoms/ButtonWithIcon';

import styles from './index.css';

interface ButtonGoogleProperty {
	className?: string;
	text?: string;
	onClick: () => void;
}

const ButtonGoogle: React.FC<ButtonGoogleProperty> = ({ className, text, onClick }) => {
	return (
		<ButtonWithIcon
			className={classnames(styles.buttonGoogle, className)}
			Icon={GoogleIcon}
			onClick={onClick}
		>
			{text}
		</ButtonWithIcon>
	);
};

export default ButtonGoogle;
