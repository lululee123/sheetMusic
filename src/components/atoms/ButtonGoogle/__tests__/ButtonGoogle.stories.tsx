import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';

import ButtonGoogle from 'components/atoms/ButtonGoogle';

const stories = storiesOf('atoms/ButtonGoogle', module);

stories.add('__interactive', () => (
	<ButtonGoogle text={text('text', 'Google 註冊')} onClick={action('onClick')} />
));
