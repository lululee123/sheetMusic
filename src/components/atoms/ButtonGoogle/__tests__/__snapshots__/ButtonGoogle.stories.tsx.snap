// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <button
    class="button buttonGoogle"
    type="button"
  >
    <test-file-stub />
    <div
      class="main"
    >
      <span>
        Google 註冊
      </span>
    </div>
  </button>
</DocumentFragment>
`;
